hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SystemClock.java

```diff
@@ -0,0 +1,49 @@
+/**
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.hadoop.util;
+
+import org.apache.hadoop.classification.InterfaceAudience.Public;
+import org.apache.hadoop.classification.InterfaceStability.Stable;
+
+/**
+ * Implementation of {@link Clock} that gives the current time from the system
+ * clock in milliseconds.
+ *
+ * NOTE: Do not use this to calculate a duration of expire or interval to sleep,
+ * because it will be broken by settimeofday. Please use {@link MonotonicClock}
+ * instead.
+ */
+@Public
+@Stable
+public final class SystemClock implements Clock {
+
+  private static final SystemClock INSTANCE = new SystemClock();
+
+  public static SystemClock getInstance() {
+    return INSTANCE;
+  }
+
+  @Deprecated
```

Comment:
Is it possible to remove this and make the constructor private? It would be unusual to introduce a new API with one of its methods immediately deprecated like this.

Reply:
Thanks for the suggestion! I think SystemClock should be a singleton, and the constructor should be private. Since this constructor is not used by any other class, I will make the constructor private.

You’re right! Introducing a new API with a deprecated method does seem unusual. I will remove the @Deprecated annotation to avoid confusion. Thanks for the review!

Reply:
The change proposed here introduces a new API in hadoop-common and advocates for callers to move to the new API. Since it is a brand new API with no pre-existing callers, we don't need to support the pre-existing interface (which was deemed deprecated).

Reply:
@cnauroth Thanks for your feedback! I understand your point that this is a new API.

I will update patch accordingly based on your suggestion.